kyla_stearns/Fundamentals/basic13.rb

```diff
@@ -0,0 +1,89 @@
+# Print 1-255
+p (1..255).to_a
+
+# Print odd numbers between 1-255
+(1..255).to_a.select { |elem| p elem if elem.odd? } 
+
+# Print Sum
+def print_sum
+	sum = 0
+	for i in (1..255)
+		p "Number: #{i} and Sum: #{sum = sum + i}"
+	end
+end
+print_sum
+
+# Iterating through an array
+x = [1,3,5,7,9,13]
+x.each { |i| p i }
+
+# Find Max
+def find_max arr
+	p arr.max 
+end
+find_max [2,4,8,1,99]
+
+# Get Average
+def avg arr
+	# to_f will account for uneven division
+	p arr.reduce(:+).to_f / arr.size
+end
+avg [2,10,3]
+
+# Array with odd numbers
+(1..255).to_a { |i| y = y.push(i) p y i.odd? }
+y = []
+(1..255).to_a.select { |elem| y.push(elem) if elem.odd? }
```

Comment:
I definitely prefer this approach, but it can be cleaned up more. On the select method if the blocks evaluates to a boolean that's all we need. Like this:
(1..255).to_a.select { |elem| elem.odd? }

.select will return a new array with the elements that return a true boolean